pkg/monitor/cluster/machineconfigpoolunmanagedcounts.go

```diff
@@ -0,0 +1,76 @@
+package cluster
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+/***
+Count the number of nodes available
+Total the nodes under machineconfigpool control
+Alert if different
+*/
+
+import (
+	"context"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func (mon *Monitor) getMachineConfigPoolNodeCounts(ctx context.Context) (int64, error) {
+	var cont string
+	var count int64
+
+	for {
+		mcps, err := mon.mcocli.MachineconfigurationV1().MachineConfigPools().List(ctx, metav1.ListOptions{Limit: 500, Continue: cont})
```

Comment:
As we use this API call 2 times now we need to add cache for it. Check 
```
ns, err := mon.listNodes(ctx)
	if err != nil {
		return 0, err
	}
```
How we do this. 

This prevent calling APi if we dont need to or we already did it

Reply:
After discussion about this, and the recommendation in the `cache.go` file saying "do not do this" (https://github.com/Azure/ARO-RP/blob/master/pkg/monitor/cluster/cache.go#L17-L18), @mjudeikis and I have decided to not use the cache and proceed with a separate call supporting the required pagination.